### PR TITLE
[RFC] llvm: allow builds against versions greater than 8

### PR DESCRIPTION
**Pull request by dvrogozh**
This patch allows builds against llvm >=8, i.e. against current master branch of https://github.com/llvm/llvm-project.git (which identifies as llvm-9).

Please, mind 2 things:
1. To build I additionally applied #55 and followed description in #54 adopted for llvm-9
2. The attached patch resolved only part of the issues (trivial issues) to build against llvm-9. I failed to resolve this kind of incompatibilities:
```
/home/dvrogozh/git/github/tmp/igc/IGC/GenISAIntrinsics/GenIntrinsics.cpp:519:83: error: no matching function for call to ‘cast<llvm::Function>(llvm::FunctionCallee)’
/home/dvrogozh/git/github/tmp/igc/IGC/Compiler/CISACodeGen/ResolveGAS.cpp:595:14: error: cannot convert ‘llvm::FunctionCallee’ to ‘llvm::Constant*’ in assignment
```

### CODE REVIEW

File: IGC/AdaptorOCL/CMakeLists.txt [hunk 1]
```diff
@@ -6,9 +6,17 @@ set(IGC_BUILD__PROJ_LABEL__AdaptorOCL "AdaptorOCL")
 include_directories("${CMAKE_CURRENT_SOURCE_DIR}")
 #include_directories("${CMAKE_CURRENT_SOURCE_DIR}/ocl_igc_shared/executable_format")
 
+set(ver LLVM_VERSION_MAJOR)
```
Comment (chivakker) [inline IGC/AdaptorOCL/CMakeLists.txt]: This has to be ${LLVM_VERSION_MAJOR}

Comment (dvrogozh) [reply IGC/AdaptorOCL/CMakeLists.txt]: yep. thank you for catching this. I eventually tried with llvm-9 and forgot other cases.

Comment (dvrogozh) [reply IGC/AdaptorOCL/CMakeLists.txt]: @chivakker : fixed. PR updated.

**Comment by dvrogozh** [comment]
This PR landed as https://github.com/intel/intel-graphics-compiler/commit/80db6d648656e89e1ca82f176c04fc77190a9410 and can be closed.

**Pull request closed without merging**